### PR TITLE
ref(api): Set access_token as GitHubClient attribute

### PR DESCRIPTION
**Pull request by slohmes**
Instead of passing the access_token in to every method call for GitHubClient, pass it into the constructor and store it as an object attribute.

### CODE REVIEW

**Comment by slohmes** [comment]
It's definitely nice to not have to pass in access token in every method call. Unfortunately,  in some places (auth pipeline, etc) the GitHubClient is initialized before the access token is available. In both those cases (FetchUser and SelectOrganization classes), I moved the client out of the init function and into the single method in the class. Since there's no other methods that'd need to access the client, it's not too big of a deal.

I'm also a little bothered that it creates a different method signatures for GitHubClient than for other auth provider clients, but I guess if maintaining a pattern there was important, the *Client objects would inherit from a parent class.

File: src/sentry/auth/providers/github/client.py
Comment (scefali) [inline src/sentry/auth/providers/github/client.py:36]: @slohmes Does `_request` not only have one argument? This seems like it would blow up.

File: src/sentry/auth/providers/github/client.py
Comment (scefali) [inline src/sentry/auth/providers/github/client.py:45]: @slohmes This is definitely a typo :)

**Comment by RyanSkonnord** [comment]
The two things that @scefali pointed out indicate a gap in unit test coverage. Before fixing them, if possible, I'd recommend adding a test that fails on those errors.

**Comment by slohmes** [comment]
Fixed and added tests -- thanks for catching!

**Review by RyanSkonnord — APPROVED** [review]
Don't know if you were waiting for an approval before merging to `github-auth`, but...

**Pull request merged**
